packages/world/src/modules/init/implementations/WorldRegistrationSystem.sol

```diff
@@ -264,27 +268,28 @@ contract WorldRegistrationSystem is System, IWorldErrors, LimitedCallContext {
     ResourceId delegationControlId,
     bytes memory initCallData
   ) public onlyDelegatecall {
-    // Store the delegation control contract address
-    UserDelegationControl._set({
-      delegator: _msgSender(),
-      delegatee: delegatee,
-      delegationControlId: delegationControlId
-    });
-
-    // If the delegation is limited...
-    if (Delegation.isLimited(delegationControlId)) {
-      // Require the delegationControl contract to implement the IDelegationControl interface
-      address delegationControl = Systems._getSystem(delegationControlId);
-      requireInterface(delegationControl, type(IDelegationControl).interfaceId);
-
-      // Call the delegation control contract's init function
-      SystemCall.callWithHooksOrRevert({
-        caller: _msgSender(),
-        systemId: delegationControlId,
-        callData: initCallData,
-        value: 0
-      });
+    createDelegation(_msgSender(), delegatee, delegationControlId, initCallData);
+  }
+
+  function registerDelegationWithSignature(
+    address delegatee,
+    ResourceId delegationControlId,
+    bytes memory initCallData,
+    address delegator,
+    bytes memory signature
```

Comment:
```suggestion
    bytes calldata signature
```

Uses slightly less gas and doesn't change the functionality. calldata can be used on params that do not get altered inside the function.

Reply:
Thanks! for some reason I thought we didn't do this in MUD but it totally works